src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -354,29 +767,61 @@ void LSHSearch<SortPolicy>::ReturnIndicesFromTable(
 
   // Compute the projection of the query in each table.
   arma::mat allProjInTables(numProj, numTablesToSearch);
+  arma::mat queryCodesNotFloored(numProj, numTablesToSearch);
   for (size_t i = 0; i < numTablesToSearch; i++)
-    //allProjInTables.unsafe_col(i) = projections[i].t() * queryPoint;
-    allProjInTables.unsafe_col(i) = projections.slice(i).t() * queryPoint;
-  allProjInTables += offsets.cols(0, numTablesToSearch - 1);
-  allProjInTables /= hashWidth;
-
-  // Compute the hash value of each key of the query into a bucket of the
-  // 'secondHashTable' using the 'secondHashWeights'.
-  arma::rowvec hashVec = secondHashWeights.t() * arma::floor(allProjInTables);
+    queryCodesNotFloored.unsafe_col(i) = projections.slice(i).t() * queryPoint;
+  queryCodesNotFloored += offsets.cols(0, numTablesToSearch - 1);
+  allProjInTables = arma::floor(queryCodesNotFloored/hashWidth);
+
+
+  // Use hashMat to store the primary probing codes and any additional codes
+  // from multiprobe LSH.
+  arma::Mat<size_t> hashMat;
+  hashMat.set_size(T + 1, numTablesToSearch);
+
+  // Compute the primary hash value of each key of the query into a bucket of
+  // the secondHashTable using the secondHashWeights.
+  hashMat.row(0) = 
+    arma::conv_to< arma::Row<size_t> >:: // floor by typecasting to size_t
+    from( secondHashWeights.t() * allProjInTables );
+  // mod to compute 2nd-level codes
+  for (size_t i = 0; i < numTablesToSearch; i++)
+    hashMat(0, i) = (hashMat(0, i) % secondHashSize);
 
-  for (size_t i = 0; i < hashVec.n_elem; i++)
-    hashVec[i] = (double) ((size_t) hashVec[i] % secondHashSize);
+  // Compute hash codes of additional probing bins
+  if (T > 0)
+  {
+    for (size_t i = 0; i < numTablesToSearch; ++i)
+    {
+      // Construct this table's probing sequence of length T.
+      arma::mat additionalProbingBins;
+      GetAdditionalProbingBins(allProjInTables.unsafe_col(i),
+                                queryCodesNotFloored.unsafe_col(i),
+                                T,
+                                additionalProbingBins);
+
+      // Map each probing bin to a bin in secondHashTable (just like we did for
+      // the primary hash table).
+      hashMat(arma::span(1, T), i) = // Compute code of rows 1:end of column i
+        arma::conv_to< arma::Col<size_t> >:: // floor by typecasting to size_t
+        from( secondHashWeights.t() * additionalProbingBins );
```

Comment:
The same comments apply from earlier; I think this can be shortened and extra spaces removed.